### PR TITLE
DD-1532: additional unit tests

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes DD-1532: additional unit tests

# Description of changes

# How to test

# Related PRs 
(Add links)
* 

# Notify
@DANS-KNAW/core-systems

### CODE REVIEW

File: src/test/java/nl/knaw/dans/layerstore/LayerDatabaseAddDirectoryTest.java [hunk 1]
```diff
@@ -163,6 +165,42 @@ public void should_add_item_records_even_if_directories_already_exist_in_another
...
+                            .layerId(1L)
+                            .path("app")
+                            .type(Item.Type.Directory)
+                            .generatedId(21L)
+                            .build();
+                    dao.saveRecords(record);
+                })
+        );
+        AssertionsForClassTypes.assertThat(e.getMessage()).startsWith("Batch update returned unexpected row count from update [0]");
+    }
```
Comment (janvanmansum) [inline src/test/java/nl/knaw/dans/layerstore/LayerDatabaseAddDirectoryTest.java:182]:
```suggestion
    }
    
```

File: src/test/java/nl/knaw/dans/layerstore/LayerDatabaseAddDirectoryTest.java [hunk 2]
```diff
@@ -163,6 +165,42 @@ public void should_add_item_records_even_if_directories_already_exist_in_another
             );
     }
 
+    @Test
```
Comment (janvanmansum) [inline src/test/java/nl/knaw/dans/layerstore/LayerDatabaseAddDirectoryTest.java:168]: Why does this generate a OLE? Isn't the database empty at the start of the test?

File: src/test/java/nl/knaw/dans/layerstore/LayerDatabaseAddDirectoryTest.java [hunk 3]
```diff
@@ -163,6 +165,42 @@ public void should_add_item_records_even_if_directories_already_exist_in_another
...
+    public void should_update_path() {
+        addToDb(1L, "james", Item.Type.Directory);
+        var r = dao.getRecordsByPath("james").get(0);
+        daoTestExtension.inTransaction(() -> {
+            r.setPath("bond");
+            dao.saveRecords(r);
+         });
+        assertThat(dao.getAllRecords().toList())
+                .usingRecursiveFieldByFieldElementComparator()
```
Comment (janvanmansum) [inline src/test/java/nl/knaw/dans/layerstore/LayerDatabaseAddDirectoryTest.java]: Leave out the `.usingRecursiveFieldByFieldElementComparator()` if you are not ignoring the generatedId field. It serves no purpose now.

**Review by janvanmansum — CHANGES_REQUESTED** [review]
Couple of minor things. See the comments. Refactorings are OK, but we also need extra tests :)

File: src/test/java/nl/knaw/dans/layerstore/LayerDatabaseAddDirectoryTest.java (hunk 2 above)
Comment (jo-pol) [reply src/test/java/nl/knaw/dans/layerstore/LayerDatabaseAddDirectoryTest.java:168]: Yes, it is empty. As the name of the test says: the `generatedId` should be in the DB. Replaced the expectations with a less obscure part of the message: "actual row count: 0; expected: 1"

**Review by janvanmansum — APPROVED** [review]

**Pull request merged**